### PR TITLE
Add cmdlets for Deploy ArubaOS CX OVA and configure

### PR DESCRIPTION
**Pull request by alagoutte**

### CODE REVIEW

File: PowerArubaCX/Public/VM.ps1 [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+
+    <#
+        .SYNOPSIS
+        Configure Password
+
+        .DESCRIPTION
+        Configure initial Password for Aruba CX OVA
+
+        .EXAMPLE
+        Set-ArubaCXVMFirtBootPassword -vmname ArubaCX -mdp toulouse
```
Comment (alagoutte) [inline PowerArubaCX/Public/VM.ps1]: Change password / mdp

File: PowerArubaCX/Public/VM.ps1 [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+
+    Process {
+
+        if (-not (Get-Command -name Set-VMKeystrokes -ErrorAction SilentlyContinue)) {
+            Throw "You need to install/import Set-VMKeystrokes script (Install-Script -Name VMKeystrokes)"
+        }
+
+        #Connection ?
+        if ( $PsBoundParameters.ContainsKey('password') ) {
+
```
Comment (alagoutte) [inline PowerArubaCX/Public/VM.ps1]: Add comment

File: PowerArubaCX/Public/VM.ps1 [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+        Set-VMKeystrokes -VMName $vmname -StringInput "interface mgmt" -ReturnCarriage $true
+        Start-Sleep 1
+        Set-VMKeystrokes -VMName $vmname -StringInput "no shutdown" -ReturnCarriage $true
+        Start-Sleep 1
+        Set-VMKeystrokes -VMName $vmname -StringInput "ip static $mgmt_ip/$mgmt_mask" -ReturnCarriage $true
+        Start-Sleep 1
+        if ( $PsBoundParameters.ContainsKey('mgmt_gateway') ) {
+            Set-VMKeystrokes -VMName $vmname -StringInput "default-gateway $mgmt_gateway" -ReturnCarriage $true
+        }
+        Start-Sleep 1
```
Comment (alagoutte) [inline PowerArubaCX/Public/VM.ps1]: Inclus start sleep on if

File: PowerArubaCX/Public/VM.ps1 [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+        }
+        #Write-Warning "You need to have a vSwitch configured on your vSphere environment even if you use a DVS"
+        #default vapp_config
+        $vapp_config = @{
+            "source" = $ovf_path
+            "name"   = $name_vm
+        }
+
+        if ($DefaultVIServers.Count -eq 0) {
+            throw "Need to be connect to vCenter (use Connect-VIServer)"
```
Comment (CedricMoreau) [inline PowerArubaCX/Public/VM.ps1]: connected

File: PowerArubaCX/Public/VM.ps1 [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
+        }
+        if (Get-VM $name_vm -ErrorAction "silentlyContinue") {
+            Throw "VM $name_vm already exist, change name or remove VM"
```
Comment (CedricMoreau) [inline PowerArubaCX/Public/VM.ps1]: exists

File: PowerArubaCX/Public/VM.ps1 [hunk 6]
```diff
@@ -0,0 +1,357 @@
...
+            }
+            else {
+                $vapp_config.add("inventory", $inventory)
+            }
+        }
+
+        $ovfConfig = Get-OvfConfiguration -Ovf $ovf_path
+
+        #Check if vSwitch is available ?
+        $ovfConfig.NetworkMapping.Null.Value = $vmnetwork
```
Comment (CedricMoreau) [inline PowerArubaCX/Public/VM.ps1:122]: Is it necessary ? We can bound directly to DVS if I remember correctly, you just can"t deploy OVF if there's no vswitch on the VMWare infrastructure

File: PowerArubaCX/Public/VM.ps1 [hunk 7]
```diff
@@ -0,0 +1,357 @@
...
+            Write-Output "$name_vm is ready to use (need to Start VM !)"
+        }
+
+    }
+
+    End {
+    }
+}
+
+function Set-ArubaCXVMFirtBootPassword {
```
Comment (CedricMoreau) [inline PowerArubaCX/Public/VM.ps1]: Firt ? For First ?

File: PowerArubaCX/Public/VM.ps1 (hunk 6 above)
Comment (alagoutte) [reply PowerArubaCX/Public/VM.ps1:122]: Yes it is the case but the idea it is to check if the vSwitch is available or not...

**Pull request merged**